### PR TITLE
Flux estimator

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request introduces the `FluxEstimator`.
So far result return is in the form of a `dict`. Do we prefer to have a `Table` row instead?
**Dear reviewer**

Some work still needed. 
The early PR should help test with the new scipy 1.4.1.

### CODE REVIEW

File: gammapy/estimators/flux_estimator.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    sigma_ul : int
+        Sigma to use for upper limit computation.
+    reoptimize : bool
+        Re-optimize other free model parameters.
+    """
+
+    def __init__(
+            self,
+            datasets,
+            source=0,
```
Comment (adonath) [inline gammapy/estimators/flux_estimator.py]: Maybe we should make the `source` argument mandatory?

File: gammapy/estimators/flux_estimator.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    def __str__(self):
+        s = f"{self.__class__.__name__}:\n"
+        s += str(self.datasets) + "\n"
+        s += str(self.model) + "\n"
+        return s
+
+    def _set_scale_model(self, datasets):
+        # set the model on all datasets
+        for dataset in datasets:
+            if len(dataset.models) > 1:
```
Comment (adonath) [inline gammapy/estimators/flux_estimator.py]: I think this `if...else` block can go just indexing into `models` should be fine.

File: gammapy/estimators/flux_estimator.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+import numpy as np
+from astropy import units as u
+from gammapy.datasets import Datasets
+from gammapy.modeling.models import ScaleSpectralModel
+from gammapy.estimators import ParameterEstimator
```
Comment (adonath) [inline gammapy/estimators/flux_estimator.py]: Use a relative import here like `from .parameter_estimator import ParameterEstimator`...

File: gammapy/estimators/flux_estimator.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+__all__ = ["FluxEstimator"]
```
Comment (adonath) [inline gammapy/estimators/flux_estimator.py]: For now I would suggest to make the `FluxEstimator` not public and keep it as an implementation detail of the `LightCurveEstimator` and `FluxPointsEstimator`, which expose the end-user API.

File: gammapy/estimators/flux_estimator.py (hunk 1 above)
Comment (registerrier) [reply gammapy/estimators/flux_estimator.py]: Done

File: gammapy/estimators/flux_estimator.py (hunk 3 above)
Comment (registerrier) [reply gammapy/estimators/flux_estimator.py]: Done, but kept absolute import

File: gammapy/estimators/flux_estimator.py (hunk 4 above)
Comment (registerrier) [reply gammapy/estimators/flux_estimator.py]: Removed `FluxEstimator` and `ParameterEstimator` from exposed API

File: gammapy/estimators/flux_estimator.py (hunk 2 above)
Comment (registerrier) [reply gammapy/estimators/flux_estimator.py]: It's gone.

**Pull request merged**